core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-admin-auth.guard.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Guard that redirects user to 401 error page
+ * if the user is not a super admin.
+ */
+
+import { Location } from '@angular/common';
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
+
+import { AppConstants } from 'app.constants';
+import { UserService } from 'services/user.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributorDashboardAdminAuthGuard implements CanActivate {
+  constructor(
+    private userService: UserService,
+    private router: Router,
+    private location: Location
+  ) {}
+
+  async canActivate(
+      route: ActivatedRouteSnapshot,
+      state: RouterStateSnapshot
+  ): Promise<boolean> {
+    const userInfo = await this.userService.getUserInfoAsync();
+    if (userInfo.isTranslationAdmin() || userInfo.isQuestionAdmin() ||
+      userInfo.isModerator()) {
+      return true;
+    }
```

Comment:
```suggestion
    if (userInfo.isTranslationAdmin() || userInfo.isQuestionAdmin()) {
      return true;
    }
```

Moderators are not allowed on cd-admin page!